openapi.yml

```diff
@@ -0,0 +1,474 @@
+        - name: survey_id
+          in: query
+          description: A UUID referring to a specific survey
+          schema:
+            type: string
+            format: uuid
+            example: 02b9c366-7397-42f7-942a-76dc5876d86d
+        - name: category
+          in: query
+          description: A string referring to what type a message is
+          schema:
+            type: string
+            enum:
+              - SURVEY
+              - TECHNICAL
+              - MISC
+            default: SURVEY
+        - name: page
+          in: query
+          description: The page of results. Used for pagination
+          schema:
+            type: integer
+            default: 1
+        - name: limit
+          in: query
+          description: The number of results to return. Can be used alone or in conjunction with `page` for pagination
+          schema:
+            type: integer
+            default: 10
+        - name: counts
+          in: query
+          description: Flag instructing the service whether to return counts about the various inboxes rOps users have
+          schema:
+            type: bool
+            default: false
+      responses:
+        200:
+          description: Successfully retrieved threads
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  total_threads:
+                    type: integer
+                    example: 1
+                  threads:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Thread'
+                  counts:
+                    type: array
+                    example:
+                      open: 2
+                      closed: 0
+                      my_conversations: 1
+                      initial_conversations: 2
+        400:
+          description: Invalid query parameters
+    post:
+      tags:
+        - Threads
+        - Existing
+      summary: Post a new thread
+      description: Create a new thread
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                category:
+                  type: string
+                  example: SURVEY
+                subject:
+                  type: string
+                  example: Help with survey
+                case_id:
+                  type: string
+                  format: uuid
+                  example: fed17f21-e662-40cc-8c3e-7f8c16607d69
+                ru_ref:
+                  type: string
+                  example: 41104403458
+                survey_id:
+                  type: string
+                  format: uuid
+                  example: 85e75391-9b91-4322-8984-3d53da29186e
+                messages:
+                  type: array
+                  items:
+                    $ref: '#/components/schemas/Message'
```

Comment:
This sends the entire message - which we won't need eventually, we will only need the body as everything else will already be in scope), but will need for the migration (if we do it by making a script that hits the endpoints to populate the data - which also means we will need the ids for the time being in the post endpoints) as we need to have all the correct data ported over

Reply:
I've left it in for now but removed it above in the POST message endpoint so we can see both options